### ISSUE TITLE
subpackage cloud.google.com/go/contextgraph/apiv1alpha1 is missing

### ISSUE THREAD

**Issue by jwendell** [istio/istio#13592]
I'm starting a new project and importing (using `dep`) istio.io@master

`dep` is complaining:
```
Solving failure: No versions of cloud.google.com/go met constraints:
	v0.37.4: Could not introduce cloud.google.com/go@v0.37.4, as its subpackage cloud.google.com/go/contextgraph/apiv1alpha1 is missing. (Package is required by istio.io/istio@master.)
```

It seems this package `cloud.google.com/go/contextgraph/apiv1alpha1` does not exist anywhere else but in istio repo, under `mixer/adapter/stackdriver` within the `vendor` folder.

Is this the right place to put dependencies? Could it get into main `vendor` dir?

**Comment by jwendell** [comment]
If I copy & paste this Gopkg.toml snippet from istio repo into mine
```
ignored = [
  # Context Graph depends on a locally vendored library that isn't (yet) available upstream.
  "istio.io/istio/mixer/adapter/stackdriver/contextgraph",
]
```

Then `dep ensure` works but the build fails:
```
# github.com/my-org/myproject/vendor/istio.io/istio/mixer/adapter/stackdriver
vendor/istio.io/istio/mixer/adapter/stackdriver/stackdriver.go:104:5: cannot use "istio.io/istio/mixer/adapter/stackdriver/contextgraph".NewBuilder(mg) (type "istio.io/istio/mixer/template/edge".HandlerBuilder) as type "github.com/my-org/myproject/vendor/istio.io/istio/mixer/template/edge".HandlerBuilder in field value:
	"istio.io/istio/mixer/template/edge".HandlerBuilder does not implement "github.com/my-org/myproject/vendor/istio.io/istio/mixer/template/edge".HandlerBuilder (wrong type for Build method)
		have Build(context.Context, "istio.io/istio/mixer/pkg/adapter".Env) ("istio.io/istio/mixer/pkg/adapter".Handler, error)
		want Build(context.Context, "github.com/my-org/myproject/vendor/istio.io/istio/mixer/pkg/adapter".Env) ("github.com/my-org/myproject/vendor/istio.io/istio/mixer/pkg/adapter".Handler, error)
vendor/istio.io/istio/mixer/adapter/stackdriver/stackdriver.go:104:31: cannot use mg (type "github.com/my-org/myproject/vendor/istio.io/istio/mixer/adapter/stackdriver/helper".MetadataGenerator) as type "istio.io/istio/mixer/adapter/stackdriver/helper".MetadataGenerator in argument to "istio.io/istio/mixer/adapter/stackdriver/contextgraph".NewBuilder:
	"github.com/my-org/myproject/vendor/istio.io/istio/mixer/adapter/stackdriver/helper".MetadataGenerator does not implement "istio.io/istio/mixer/adapter/stackdriver/helper".MetadataGenerator (wrong type for GenerateMetadata method)
		have GenerateMetadata() "github.com/my-org/myproject/vendor/istio.io/istio/mixer/adapter/stackdriver/helper".Metadata
		want GenerateMetadata() "istio.io/istio/mixer/adapter/stackdriver/helper".Metadata
```

**Comment by jwendell** [comment]
@jeffmendoza any hints?

**Comment by jeffmendoza** [comment]
I'm not a dep expert. We had to import this client library manually here:
https://github.com/istio/istio/tree/master/mixer/adapter/stackdriver/vendor
As it's not available in it's proper locations yet.

**Comment by jwendell** [comment]
@jeffmendoza yeah, thanks for your response. I saw that you added it that's why I guessed you might give some light...

**Comment by jeffmendoza** [comment]
Again, not sure on the dep capabilities, but I think that this can't go in the top level vendor, as  dep will delete it because it is not upstream.

**Comment by jwendell** [comment]
fixed by c2d324a2e7f430b4d002acbf17ac06127c6be741
